src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/net/GRPCConnectionManager.java

```diff
@@ -54,17 +55,15 @@
   private File trustedCasFile;
 
   /**
-   * Map of remote host to a Netty channel to that host.
+   * Map of remote hostId to a Netty channel to that host.
    */
-  private ConcurrentMap<String, AtomicReference<ManagedChannel>> perHostChannelMap =
-      new ConcurrentHashMap<>();
+  private ConcurrentMap<InstanceDetails.Id, AtomicReference<ManagedChannel>> perHostChannelMap = new ConcurrentHashMap<>();
```

Comment:
I agree with most of the refactor points mentioned in the description of the PR, however, can you explain why we need to use node ids instead of ip+port combo? It would've achieved the same goal, unless I'm missing something.

Reply:
You are right, ip+port will have the same affect. I used the node-id as it is a parameter, on its own (without combining with anything else), is guaranteed to be unique in a cluster.

Reply:
Thanks for answering my question. I'm just thinking about the usecases where ip+port wouldn't change but node id would and if such scenarios are handled in the code if they were not handled in the past. Will take a look again.

Reply:
makes sense. Please take a look.